src/main/java/service/impl/FruitTransactionServiceImpl.java

```diff
@@ -0,0 +1,38 @@
+package service.impl;
+
+import db.Storage;
+import model.FruitTransaction;
+import service.FruitTransactionService;
+import strategy.FruitTransactionStrategy;
+
+public class FruitTransactionServiceImpl implements FruitTransactionService {
+    private static final int OPERATION_INDEX = 0;
+    private static final int FRUIT_NAME_INDEX = 1;
+    private static final int QUANTITY_INDEX = 2;
+    private FruitTransactionStrategy strategy;
+    private Storage fruitsStorage;
+
+    public FruitTransactionServiceImpl(FruitTransactionStrategy strategy, Storage fruitsStorage) {
+        this.strategy = strategy;
+        this.fruitsStorage = fruitsStorage;
+    }
+
+    public void addToStorage(String[] dataFromFile) {
```

Comment:
This method responsible for parsing String to FruitTransaction class, for choosing operationHandler and saving data to Storage... it violates SPR. Let's split logic of this method into several methods